src/main/java/core/basesyntax/service/shopservice/ShopServiceImpl.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax.service.shopservice;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.strategy.OperationHandler;
+import core.basesyntax.service.strategy.OperationStrategy;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
```

Comment:
The class ShopServiceImpl is well-defined and follows the class naming convention as per the checklist. It also seems to be focused on a single responsibility, adhering to the SOLID principles.